snap/snapcraft.yaml

```diff
@@ -12,256 +13,318 @@ description: |
-    plugs: [network, network-bind]
-    daemon: forking
+    adapter: none
+    after: [consul]
+    command: >-
+      bin/config-seed
+      --cmd $SNAP_DATA/config
+      -confdir $SNAP_DATA/config/config-seed/res
+      --props $SNAP_DATA/config/config-seed/res/properties
+    daemon: oneshot
     passthrough:
-      after:
-        - consul
+      start-timeout: 15m
+    plugs: [network, network-bind]
+  redis:
+    adapter: full
+    after: [core-config-seed]
+    command: bin/redis-server $SNAP_DATA/redis/redis.conf
+    daemon: simple
+    plugs: [network, network-bind]
   mongod:
-    command: bin/mongod-wrapper.sh
-    stop-command: bin/mongod --shutdown --dbpath ${SNAP_DATA}/mongo/db
-    plugs: [hardware-observe, network, network-bind, system-observe]
+    adapter: full
+    after: [core-config-seed]
+    command: bin/mongod --dbpath $SNAP_DATA/mongo/db --logpath $SNAP_COMMON/mongodb.log --smallfiles
     daemon: simple
-    passthrough:
-      after:
-        - core-config-seed
+    plugs: [hardware-observe, network, network-bind, system-observe]
+    stop-command: bin/mongod --shutdown --dbpath $SNAP_DATA/mongo/db
   cassandra:
+    adapter: full
     command: bin/cassandra-wrapper.sh
     daemon: forking
+    passthrough:
+      start-timeout: 15m
     plugs:
       - network
       - network-bind
       - mount-observe
   kong-daemon:
+    adapter: full
+    after:
+      - cassandra
+      - pkisetup
     command: bin/kong-daemon.sh
+    daemon: forking
     environment:
       KONG_CONF: $SNAP_DATA/config/security-api-gateway/kong.conf
       KONG_PROXY_ACCESS_LOG: $SNAP_COMMON/logs/kong-proxy-access.log
       KONG_ADMIN_ACCESS_LOG: $SNAP_COMMON/logs/kong-admin-access.log
       KONG_PROXY_ERROR_LOG: $SNAP_COMMON/logs/kong-proxy-error.log
       KONG_ADMIN_ERROR_LOG: $SNAP_COMMON/logs/kong-admin-error.log
       KONG_ADMIN_LISTEN: "0.0.0.0:8001, 0.0.0.0:8444 ssl"
-    stop-command: bin/kong-stop.sh
-    daemon: forking
+    passthrough:
+      start-timeout: 15m
     plugs:
       - network
       - network-bind
-    passthrough:
-      after:
-        - cassandra
-        - pkisetup
+    stop-command: bin/kong-stop.sh
   pkisetup:
+    adapter: full
     command: bin/pkisetup-wrapper.sh
+    daemon: oneshot
     environment:
       PKI_SETUP_VAULT_FILE: $SNAP_DATA/config/security-secret-store/pkisetup-vault.json
       PKI_SETUP_KONG_FILE: $SNAP_DATA/config/security-secret-store/pkisetup-kong.json
-    daemon: oneshot
+    passthrough:
+      start-timeout: 15m
   vault:
-    command: bin/vault server --config="$VAULT_CONFIG"
+    adapter: none
+    after:
+      - consul
+      - pkisetup
+      - core-config-seed
+    command: bin/vault server --config $VAULT_CONFIG
+    daemon: simple
     environment:
-      VAULT_CONFIG: "$SNAP_DATA/config/security-secret-store/vault-config.json"
+      VAULT_CONFIG: "$SNAP_DATA/config/security-secret-store/vault-config.hcl"
       VAULT_ADDR: "https://localhost:8200"
```

Comment:
Hmm, this means vault is always bound to port 8200 in the snap?

Reply:
Yes but this is no different than it was before. There's not currently a clean way to modify environment variables for services so if we wanted to expose the port number here to say `snap set`, we would need to implement our own wrapper which doesn't use the `environment` section here which is what I was trying to avoid with many of these changes.

Reply:
Yes, I realize this wasn't change...  Might be worth an issue to track.

Reply:
Filed as https://github.com/edgexfoundry/edgex-go/issues/1387